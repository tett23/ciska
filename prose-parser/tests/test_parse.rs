extern crate prose_parser;

use insta::assert_yaml_snapshot;
use std::{fs, path::Path};

#[test]
fn parse_empty() {
    assert_yaml_snapshot!(prose_parser::parse(""), @r###"
    ---
    Ok:
      name: document
    "###)
}

#[test]
fn parse() {
    read_fixtures().iter().for_each(|(name, content)| {
        assert_yaml_snapshot!(name.as_str(), prose_parser::parse(content.as_str()));
    })
}

fn read_fixtures() -> Vec<(String, String)> {
    fs::read_dir("./tests/fixtures")
        .unwrap()
        .filter_map(|item| {
            let ent = item.ok()?;

            match ent.file_type().ok()?.is_file() {
                true => Some(ent.path().to_string_lossy().into_owned()),
                false => None,
            }
        })
        .map(|item| match fs::read_to_string(&item) {
            Ok(ent) => Ok((
                Path::new(&item)
                    .file_stem()
                    .unwrap()
                    .to_string_lossy()
                    .to_string(),
                ent,
            )),
            Err(e) => Err(e),
        })
        .flat_map(|item| item)
        .collect::<_>()
}
