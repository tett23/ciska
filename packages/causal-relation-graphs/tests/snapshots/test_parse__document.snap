---
source: prose-parser/tests/test_parse.rs
assertion_line: 51
expression: "prose_parser::parse(content.as_str()).unwrap()"

---
name: document
children:
  - name: heading
    children:
      - name: heading_begin
      - name: text
        value: foo
  - name: line_break
  - name: paragraph
    children:
      - name: text
        value: bar
  - name: line_break
  - name: block_comment
    children:
      - name: text
        value: block comment
  - name: line_break
  - name: paragraph
    children:
      - name: tag
        children:
          - name: text
            value: tag-name
      - name: tag
        children:
          - name: text
            value: tag2
      - name: text
        value: "paragraph text "
      - name: inline_comment
        children:
          - name: text
            value: inline comment
  - name: line_break

